### PR TITLE
Introduce C struct for waypoint

### PR DESCRIPTION
**Pull request by blmarket**
Introduce nox_waypoint_t struct

Some of the script code will be too complicated without proper
definition of the waypoint struct.

Struct size and field definitions were inferred from code
reading/writing nox_xxx_waypointsHead_2523752, and game save/load logic
in GAME4.c, so some fields can be incorrect.

## Required sign-off

- [x] I confirm that my PR does not contain any commercial or protected assets and/or source code.
- [x] I agree in advance that my codes will be licensed automatically under the Apache License or similar BSD/MIT-like
      open source licenses in case if OpenNox Project will adopt such a non-GPL license in the future.

### CODE REVIEW

File: src/waypoint.go [hunk 1]
```diff
@@ -17,7 +18,7 @@ func asWaypoint(p unsafe.Pointer) *Waypoint {
 }
 
 func (s *Server) firstWaypoint() *Waypoint {
-	return asWaypoint(unsafe.Pointer(uintptr(C.nox_xxx_waypointsHead_2523752)))
+	return (*Waypoint)(C.nox_xxx_waypointsHead_2523752)
```
Comment (dennwc) [inline src/waypoint.go]: Please use `asWaypoint` here. It helps determine later if it's safe to change struct layout and completely move it to Go (if `asWaypoint` and `Waypoint.C` are not called).

Similar to objects, you could define `func asWaypoint(p *nox_waypoint_t) *Waypoint {` and `func asWaypointP(p unsafe.Pointer) *Waypoint {`, so both raw pointer and C struct pointers can be passed conveniently.

File: src/defs.h [hunk 2]
```diff
@@ -2035,4 +2035,28 @@ typedef struct nox_gui_server_ent_t {
...
+
+typedef struct nox_waypoint_t {
+	uint32_t ind;                  // 0, 0
+	uint32_t field_1;              // 1, 4
+	float x;                       // 2, 8
+	float y;                       // 3, 12
+	char name[76];                 // 4, 16
+	nox_waypoint_inner points[32]; // 5, 92
+	uint32_t field_6[32];          // 6, 348
+	uint8_t points_cnt;            // 7, 476, padded to 4 bytes
```
Comment (dennwc) [inline src/defs.h]: Please add explicit padding (a few unused fields afterwards).

File: src/waypoint.go (hunk 1 above)
Comment (blmarket) [reply src/waypoint.go]: :+1:

File: src/defs.h (hunk 2 above)
Comment (blmarket) [reply src/defs.h]: :heavy_check_mark:

**Comment by blmarket** [comment]
Note: #534 happened to include changes in this PR. If that PR looks fine, please close this one and just merge that.

**Comment by dennwc** [comment]
OK, thank you, closing this one then. I will also rebase #534, I found a severe bug in the current `dev`, so it's better to roll it back a bit.

**Pull request closed without merging**